diff --git a/scripts/build.sh b/scripts/build.sh
index 2606594..0dc1c29 100644
--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -8,14 +8,6 @@
 # INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
 # See LICENSE in the root of the software repository for the full text of the License.
 # -----------------------------------------------------------------------------------------------------------
-if [ -n "$ASCEND_HOME_PATH" ]; then
-    _ASCEND_INSTALL_PATH=$ASCEND_HOME_PATH
-fi
-
-export ASCEND_TOOLKIT_HOME=${_ASCEND_INSTALL_PATH}
-export ASCEND_HOME_PATH=${_ASCEND_INSTALL_PATH}
-
-source "$(dirname "$_ASCEND_INSTALL_PATH")/set_env.sh"
 
 CURRENT_DIR=$(pwd)
 SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4e1de15..fa37a89 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -86,16 +86,18 @@ install(TARGETS shmem_utils
     LIBRARY DESTINATION lib
 )
 
-add_library(shmem SHARED $<TARGET_OBJECTS:aclshmem_device> $<TARGET_OBJECTS:aclshmem_host>)
-target_link_options(shmem PRIVATE --cce-fatobj-link)
-target_link_libraries(shmem PRIVATE shmem_utils)
+add_library(hpshmem SHARED $<TARGET_OBJECTS:aclshmem_device> $<TARGET_OBJECTS:aclshmem_host>)
+target_link_options(hpshmem PRIVATE --cce-fatobj-link)
+set_target_properties(hpshmem PROPERTIES
+        INSTALL_RPATH $ORIGIN)
+target_link_libraries(hpshmem PRIVATE shmem_utils)
 
 # MPI
 if(ACLSHMEM_MPI_SUPPORT)
     separate_arguments(ACLSHMEM_CXX_LINK_FLAGS NATIVE_COMMAND "${MPI_CXX_LINK_FLAGS}")
-    target_link_options(shmem INTERFACE ${ACLSHMEM_CXX_LINK_FLAGS})
-    target_compile_definitions(shmem INTERFACE ${MPI_CXX_COMPILE_DEFINITIONS})
-    target_compile_options(shmem INTERFACE ${MPI_CXX_COMPILE_OPTIONS})
+    target_link_options(hpshmem INTERFACE ${ACLSHMEM_CXX_LINK_FLAGS})
+    target_compile_definitions(hpshmem INTERFACE ${MPI_CXX_COMPILE_DEFINITIONS})
+    target_compile_options(hpshmem INTERFACE ${MPI_CXX_COMPILE_OPTIONS})
 
     add_library(aclshmem_bootstrap_mpi SHARED)
     target_sources(aclshmem_bootstrap_mpi PRIVATE host/bootstrap/shmemi_bootstrap_mpi.cpp)
@@ -157,7 +159,7 @@ install(TARGETS aclshmem_bootstrap_config_store
         LIBRARY DESTINATION lib)
 
 # 安装配置
-install(TARGETS shmem
+install(TARGETS hpshmem
     LIBRARY DESTINATION lib
     PUBLIC_HEADER DESTINATION include
 )
diff --git a/src/host/utils/shmemi_num_util.h b/src/host/utils/shmemi_num_util.h
index 880b32a..6909092 100644
--- a/src/host/utils/shmemi_num_util.h
+++ b/src/host/utils/shmemi_num_util.h
@@ -71,13 +71,13 @@ public:
     */
     static uint32_t ExtractBits(uint32_t flag, uint8_t startBit, uint8_t bitLength)
     {
-        if (startBit >= UINT32_WIDTH) {
+        if (startBit >= 32) {
             return UINT32_MAX;
         }
-        if (bitLength == 0 || bitLength >= UINT32_WIDTH) {
+        if (bitLength == 0 || bitLength >= 32) {
             return UINT32_MAX;
         }
-        if (startBit + bitLength > UINT32_WIDTH) {
+        if (startBit + bitLength > 32) {
             return UINT32_MAX;
         }

